src/app/clusters/user-label-server/user-label-server.cpp

```diff
@@ -151,7 +151,14 @@ CHIP_ERROR UserLabelAttrAccess::WriteLabelList(const ConcreteDataAttributePath &
         ReturnErrorOnFailure(aDecoder.Decode(entry));
         VerifyOrReturnError(IsValidLabelEntry(entry), CHIP_IM_GLOBAL_STATUS(ConstraintError));
 
-        return provider->AppendUserLabel(endpoint, entry);
+        // Append the single user label entry
+        CHIP_ERROR err = provider->AppendUserLabel(endpoint, entry);
+        if (err == CHIP_ERROR_NO_MEMORY)
+        {
+            return CHIP_IM_GLOBAL_STATUS(ResourceExhausted);
+        }
+
+        return err;
```

Comment:
Please test that the storage is actually released when the list goes down in size